.github/workflows/e3sm-gh-pages.yml

```diff
@@ -12,28 +12,46 @@ on:
 concurrency:
   # Prevent 2+ copies of this workflow from running concurrently
   group: e3sm-docs-action
-  
+
 jobs:
-  Build-and-Deploy-docs:
+  e3sm-docs:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v4
+      - 
+        name: Check out code
+        uses: actions/checkout@v4
         with:
           show-progress: false
-          fetch-depth: 0 # Needed, or else gh-pages won't be fetched, and push rejected
+          # We will need submodules when/if we build EAMxx docs here
           submodules: false # speeds up clone and not building anything in submodules
-      - name: Show action trigger
+      - 
+        name: Show action trigger
         run: echo "= The job was automatically triggered by a ${{github.event_name}} event."
-      - name: Set up Python 3.10
+      - 
+        name: Set up Python 3.10
         uses: actions/setup-python@v4.7.0
         with:
           python-version: "3.10"
-      - name: Install python deps
+      - 
+        name: Install Python deps
         run: python3 -m pip install mkdocs-material pymdown-extensions mkdocs-monorepo-plugin mdutils
-      # build every time (PR or push to master)
-      - name: Build
+      - 
+        name: Build
         run: mkdocs build --strict --verbose
-      # deploy only when it is a push
-      - if: ${{ github.event_name == 'push' }}
-        name: Deploy
-        run: mkdocs gh-deploy
+      - 
+        # Only deploy to the main github page when there is a push to master
+        if: ${{ github.event_name == 'push' }}
+        name: GitHub Pages action
+        uses: JamesIves/github-pages-deploy-action@v4
+        with:
+          # Do not remove existing pr-preview pages
+          clean-exclude: pr-preview
+          folder: ./site/
+      - 
+        # If it's a PR from within the same repo, deploy to a preview page
+        # For security reasons, PRs from forks cannot write into gh-pages for now
```

Comment:
Can we still do a build check of docs in PRs from forks?

Reply:
Currently, no. At least, not through this action. The documentation on the pr-preview-action github page claims it will support PRs from fork in v2. Although, that message on their docs is 1.5y old, so I'm not 100% confident they'll follow through on that...

That said, it's not complicate to write our own actions. We could easily fork that action, upgrade it to handle forks, and store it on e3sm-project. In the long run, it may even be better, in case actions are not maintained...

Reply:
The current action will (I think) check the doc build on a PR from anywhere.  I'd like to keep that.

Reply:
It will definitely build the docs, so you'll know if there's some error in the docs generation, but I think the preview will fail. In order to preview it, it has to push on the gh-pages branch of our repo, which cannot be done if the PR comes from a fork.

Reply:
Ah, wait, you mean the action currently in master? I think the current master version is not doing any preview for PRs. The deploy step is guarded by `if: ${{ github.event_name == 'push' }}`.

Reply:
Its not doing a preview but its building for every PR even from forks.  I want to keep build-from-fork.  Sounds like it will.